squirrel.worker/src/main/java/org/dice_research/squirrel/encoder/TripleEncoder.java

```diff
@@ -0,0 +1,102 @@
+package org.dice_research.squirrel.encoder;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URISyntaxException;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.graph.Triple;
+import org.apache.jena.util.URIref;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class that can encode triples to UTF8
+ * 
+ * @author Geraldo de Souza Junior gsjunior@mail.uni-paderborn.de
+ *
+ */
+public class TripleEncoder {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(TripleEncoder.class);
+	private static TripleEncoder tripleEncoder;
```

Comment:
Why does the TripleEncoder implement the Singleton pattern? Is that really necessary?

Reply:
I wanted to avoid unnecessary instantiations of this class across the JVM.  This utility class can be shared by the entire application, so its a good way to manage resources

Reply:
No. Depending on our goals, there are better ways.

1. We want to use the TripleEncoder wherever we encode triples and we want to reduce the amount of resources that costs. In that case, `static` methods are better than instantiating the class.
2. We want to use the TripleEncoder as a default implementation that can be exchanged if another implementation is necessary. In that case, using a singleton will reduce our ability to exchange it.

So in both cases, other options are better :thinking:

Reply:
I left the Triple encoder instance on AbstractAnalzyer, without the Singleton Pattern.